gitlabform/gitlabform/processors/group/group_shared_with_processor.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import sys
+
+import cli_ui
+
+from gitlabform.gitlab import GitLab
+from gitlabform.gitlab.core import NotFoundException
+from gitlabform import EXIT_INVALID_INPUT
+from gitlabform.gitlabform.processors.abstract_processor import AbstractProcessor
+
+
+class GroupSharedWithProcessor(AbstractProcessor):
+    def __init__(self, gitlab: GitLab):
+        super().__init__("group_shared_with")
+        self.gitlab = gitlab
+
+    def _process_configuration(self, group: str, configuration: dict):
+        groups_to_set_by_groupname = configuration.get("group_shared_with")
+        print(groups_to_set_by_groupname)
+        if groups_to_set_by_groupname is None:
+            groups_to_set_by_groupname = {}
+
+        # group users before by group name
+        groups_before = self.gitlab.get_group_case_insensitive(group)[
+            "shared_with_groups"
+        ]
+        logging.debug("Group shared with BEFORE: %s", groups_before)
+        groups_before_by_group_id = dict()
+        for share_details in groups_before:
+            groups_before_by_group_id[share_details["group_id"]] = share_details
+
+        groups_to_set_by_group_id = set()
+
+        for groupname in groups_to_set_by_groupname:
+
+            try:
+                group_id = self.gitlab.get_group_case_insensitive(groupname)["id"]
+            except NotFoundException:
+                cli_ui.error(f"Group {groupname} not found.")
+                sys.exit(EXIT_INVALID_INPUT)
```

Comment:
The pattern in our current codebase for this is to convert names to ids inside the `gitlabform.gitlab` package, see https://github.com/egnyte/gitlabform/blob/d926f5561007d82c88e1a242c927993527a61c9f/gitlabform/gitlab/projects.py#L229-L236 as an example. I think that we should do the same thing here.

Reply:
I guess that this is done this way for performance reasons, to not re-do the same get group id requests over and over, and that makes sense, but I propose to cover this by introducing cache to wrap around some methods that we know that are safe to cache, such as those getting object ids.

Reply:
WDYT?

Reply:
Yes, I was worried about the performance of many requests to convert the group name into an id. I've changed this now so the api uses group names, and internally caches the mapping to an id. Please let me know if this looks ok now.